core/templates/dev/head/components/forms/schema_editors/SchemaBasedListEditorDirective.js

```diff
@@ -135,12 +135,23 @@ oppia.directive('schemaBasedListEditor', [
             }
           };
 
+          var _deleteElementIfEmpty = function() {
+            for (var i = 0; i < $scope.localValue.length - 1; i++) {
+              if ($scope.localValue[i].length === 0) {
+                $scope.deleteElement(i);
+                i--;
+              }
+            }
+          };
+
           $scope.lastElementOnBlur = function() {
             _deleteLastElementIfUndefined();
+            _deleteElementIfEmpty();
             $scope.showAddItemButton();
           };
 
           $scope.showAddItemButton = function() {
+            _deleteElementIfEmpty();
```

Comment:
Hang on -- I agree with the function being here. I'm just saying it's a no-op in the other method, if I understand it correctly.

The question I asked about this function was whether it should only handle deletion of the field that was just blurred, rather than all fields.